dialogue-futures/src/test/java/com/palantir/dialogue/futures/DialogueFuturesTest.java

```diff
@@ -0,0 +1,216 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.futures;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import java.util.concurrent.CancellationException;
+import org.junit.jupiter.api.Test;
+
+class DialogueFuturesTest {
+
+    @Test
+    void testTransform_success() throws Exception {
+        SettableFuture<String> original = SettableFuture.create();
+        ListenableFuture<String> doubled = DialogueFutures.transform(original, value -> value + value);
+        original.set("a");
+        assertThat(doubled.get()).isEqualTo("aa");
+    }
+
+    @Test
+    void testTransform_originalCancel() {
+        SettableFuture<String> original = SettableFuture.create();
+        ListenableFuture<String> doubled = DialogueFutures.transform(original, value -> value + value);
+        assertThat(doubled.cancel(false)).isTrue();
+        assertThat(doubled).isCancelled();
+        assertThatThrownBy(doubled::get).isInstanceOf(CancellationException.class);
+    }
+
+    @Test
+    void testTransform_resultCancel() {
+        SettableFuture<String> original = SettableFuture.create();
+        ListenableFuture<String> doubled = DialogueFutures.transform(original, value -> value + value);
+        assertThat(doubled.cancel(false)).isTrue();
+        assertThat(doubled).isCancelled();
+        assertThatThrownBy(doubled::get).isInstanceOf(CancellationException.class);
+    }
+
+    @Test
+    void testTransformAsync_success() throws Exception {
+        SettableFuture<String> one = SettableFuture.create();
+        SettableFuture<String> two = SettableFuture.create();
+        ListenableFuture<String> transformed = DialogueFutures.transformAsync(one, _value -> two);
+        assertThat(transformed).isNotDone();
+        one.set("a");
+        assertThat(transformed).isNotDone();
+        two.set("b");
+        assertThat(transformed).isDone();
+        assertThat(transformed.get()).isEqualTo("b");
+    }
+
+    @Test
+    void testTransformAsync_originalCancel() {
+        SettableFuture<String> one = SettableFuture.create();
+        SettableFuture<String> two = SettableFuture.create();
+        ListenableFuture<String> transformed = DialogueFutures.transformAsync(one, _value -> two);
+        assertThat(transformed).isNotDone();
+        assertThat(one.cancel(false)).isTrue();
+        assertThat(transformed).isCancelled();
+        assertThat(transformed).isDone();
+        assertThatThrownBy(transformed::get).isInstanceOf(CancellationException.class);
+        // The intermediate future is not impacted
```

Comment:
i'd pop helpful little explanations like this into a `.describedAs()` on the assertion below